### PR TITLE
Fix/unreadable device

### PR DESCRIPTION
**Pull request by szymonlesisz**
Distinguish unreadable device from unacquired device.
Unreadable device error (reason) is passed in unreadable Device.id field.

Device selection is required if unreadable device is present (popup mode)

### CODE REVIEW

File: src/js/env/browser/browserUtils.js [hunk 1]
```diff
@@ -51,6 +51,12 @@ export const getBrowserState = (supportedBrowsers: {
     };
 };
 
+export const getOS = () => {
+    if (!navigator || !navigator.userAgent) return 'unknown';
+    const browser = Bowser.getParser(navigator.userAgent);
+    return browser.getOS().name.toLowerCase();
```
Comment (matejkriz) [inline src/js/env/browser/browserUtils.js]: `name` is defined as possibly undefined (probably could never happen, but better safe than sorry)

Comment (szymonlesisz) [reply src/js/env/browser/browserUtils.js]: addressed in 4270cee

also a lot of tests were added

File: src/ts/types/__tests__/index.ts [hunk 2]
```diff
@@ -87,6 +87,13 @@ export const events = async () => {
             payload.features.unfinished_backup;
             payload.features.no_backup;
             payload.features.model;
+            // @ts-ignore: error does not exist
+            payload.error.toLowerCase();
```
Comment (matejkriz) [inline src/ts/types/__tests__/index.ts]: maybe this would work in TS file?
```suggestions
payload.error?.toLowerCase();
```

**Review by matejkriz — APPROVED** [review]

File: src/ts/types/__tests__/index.ts (hunk 2 above)
Comment (szymonlesisz) [reply src/ts/types/__tests__/index.ts]: `@ts-expect-error` it is, thanks

**Pull request merged**